### PR TITLE
store: implement add/delete/query for subscription

### PR DESCRIPTION
**Pull request by tailingchen**

### CODE REVIEW

File: model/query.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// You should have received a copy of the GNU Lesser General Public License
+// along with the eth-indexer library. If not, see <http://www.gnu.org/licenses/>.
+
+package model
+
+import (
+	"fmt"
+)
+
+type QueryConfig struct {
```
Comment (jeddie-yhc) [inline model/query.go]: How about rename to `QueryParameters`? We read these parameters from the request.

Comment (tailingchen) [reply model/query.go]: Good! Updated in new version.

File: store/subscription/subscription.go [hunk 2]
```diff
@@ -111,3 +114,27 @@ func (t *store) Reset(from, to int64) error {
...
+	var result []*model.Subscription
+	err = db.Offset(start).Limit(query.Limit).Order(query.OrderString()).Find(&result).Error
+	if err != nil {
+		return nil, 0, err
+	}
+	return result, total, nil
+}
+
+func (t *store) BatchDelete(groupID int64, addrs [][]byte) (err error) {
+	db := t.db.Where("`group` = ?", groupID).Where("address in (?)", addrs)
```
Comment (markya0616) [inline store/subscription/subscription.go]: How about use AND? 
```
t.db.Where("`group` = ? AND address in (?)", groupID, addrs)
```

File: store/subscription/subscription.go [hunk 3]
```diff
@@ -111,3 +114,27 @@ func (t *store) Reset(from, to int64) error {
...
+	return db.Delete(&model.Subscription{}).Error
```
Comment (markya0616) [inline store/subscription/subscription.go]: Can you add a test if there are many subscriptions with different groups and only deleted subscription with the given group?

File: store/subscription/subscription.go (hunk 2 above)
Comment (tailingchen) [reply store/subscription/subscription.go]: `Where ` will combined as `AND`.
So
```
 t.db.Where("`group` = ?", groupID).Where("address in (?)", addrs)
```
is equal to 
```
t.db.Where("`group` = ? AND address in (?)", groupID, addrs)
```

File: store/subscription/subscription.go (hunk 3 above)
Comment (tailingchen) [reply store/subscription/subscription.go]: After discuss with @kimmylin, we don't support delete subscriptions for now.

**Pull request merged**